notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/TagEnricher.java

```diff
@@ -10,27 +16,32 @@
  */
 public class TagEnricher implements InstallationEnricher, Tagable {
 
-    public final Set<String> mTags;
+    private SharedPreferences mPreferences;
 
     /**
      * Creates an empty TagEnricher.
      */
     public TagEnricher() {
-        mTags = new HashSet<String>();
+
     }
 
     /**
      * Creates a TagEnricher with a pre-populated set of tags to apply.
      * @param tags The initial set of tags that should be applied to future {@link Installation}s.
      */
-    public TagEnricher(Collection<? extends String> tags) {
+    public TagEnricher(Context context, Collection<? extends String> tags) {
         this();
-        mTags.addAll(tags);
+        SetPreferences(context);
+        addTags(tags);
+    }
+
+    public void SetPreferences (Context context) {
+        mPreferences = context.getSharedPreferences(String.valueOf(R.string.preference_file_key), Context.MODE_PRIVATE);
     }
 
     @Override
     public void enrichInstallation(Installation subject) {
-        subject.addTags(mTags);
+        subject.addTags((Collection<String>) getTags());

```

Comment:
I mention it elsewhere, but this is another example of downcasting. I'd prefer to just go ahead and use something like `subject.addTags(mPreferences.getStringSet("tags"))`.